### DISCUSSION TITLE
How do we consolidate lazer builds potentially having undeployed SR/PP changes in the future?

### DISCUSSION THREAD

**Discussion by apollo-dw** [ppy/osu#24747] [Ideas]
I don't think it's a good idea to show undeployed/staged changes to difficulty or performance to users. Either the UX needs to change to make it super clear, or SR retrieval for submitted beatmaps needs to be moved to online.

**Comment by peppy** [comment]
For now we don't really care, all changes will be deployed at once when we do the leaderboard reset.

In the future we'll likely need to maintain a separate non-`master` branch for diffcalc changes.

**Comment by Rekunan** [comment]
I've brought this up in the past before, somewhat similarly: #19724
